### DISCUSSION TITLE
APIConnectionError  on Python 3.11.9,  SK 1.2.0

### DISCUSSION THREAD

**Discussion by giobb** [microsoft/semantic-kernel#9274] [Q&A] [answered]
Hi everyone,

I need help on this one. I have a similar C# codes that works just fine using the same settings and Managed Identity. Initially I thought MI was the culprit but switching to key does not help either.

![image](https://github.com/user-attachments/assets/adb72ad1-be53-4242-aa06-e3b485cf0b55)

Here is the entire block. I have applied the trick on using _base_url_ instead of of _endpoint_ from another thread discussing same error. Not doing so results in 404. 

![image](https://github.com/user-attachments/assets/d9aa7771-1785-4900-b78d-c811fc89dd03)

**Comment by moonbox3** [comment, marked as answer]
Hello, @giobb, thanks for reaching out.

You'll want to configure your `endpoint` to be: `https://certain-ep.openai.azure.com/` (without the `/openai/`). Additionally, please use the `endpoint` keyword argument, instead of the `base_url` keyword argument. The `base_url` keyword argument is assuming you're passing in the `base_url` like: `https://certain-ep.openai.azure.com/openai/deployments/gpt-4o`.

As part of initializing the AzureChatCompletion service, we create the `base_url` from the `endpoint` and the `chat_deployment_name` such as:

`https://certain-ep.openai.azure.com/openai/deployments/gpt-4o`

**Reply by moonbox3** [reply]
@giobb, one debugging tip for the future is if you enable debug level logging, you'll be able to see the API call made, and would be able to then see that the URI used to make the API call to Azure OpenAI could be misconfigured. 

In your script you can add:

```
import logging

logging.basicConfig(level=logging.DEBUG)
```

In the logs/console, you'd see something like:

```
Request: POST https://certain-ep.openai.azure.com/openai/openai/deployments/gpt-4o/chat/completions?api-version=2024-05-01-preview
```

Hope that helps if you run into any `404` issues later on.

**Comment by giobb** [comment]
@moonbox3 Thanks for the time and suggestion. Using _endpoint_ (without /openai) with _ad_token_ or _api_key_ but still I get the same issue.  Could this be firewall, other network issues?

**Reply by moonbox3** [reply]
Hi @giobb, I am using the same code as you and I am not experiencing any issues -- meaning, I am passing in my endpoint, and an ad_token. Can you confirm that you have the appropriate role on your Azure OpenAI resource? See [here](https://github.com/microsoft/semantic-kernel/tree/main/python/samples/concepts#microsoft-entra-token-authentication).

```python
chat_service = AzureChatCompletion(
    service_id=service_id,
    endpoint="https://<resource>.openai.azure.com/",
    ad_token=ad_token,
)
kernel.add_service(chat_service)
```

If you do have the correct role, and you're getting a 404 still, can you confirm via debug logging that your POST request is going to the properly formatted final endpoint? Similar to what I pasted above?

```
Request: POST https://<resource>.openai.azure.com/openai/deployments/gpt-4o/chat/completions?api-version=2024-05-01-preview
```

**Reply by giobb** [reply]
Yup just double-checked. I tried even with api key. I have a similar construct in C# and it's working fine. That's the most frustrating part.  

![image](https://github.com/user-attachments/assets/c948ddfb-ca87-439b-ad76-e6cc6c258cbb)

**Reply by giobb** [reply]
We'll try to upgrade to the latest SK version as you suggested. Thanks again.

**Reply by moonbox3** [reply]
For context: this was solved with an updated api version to the correct format (YYYY-MM-DD) like `2024-06-01`.
